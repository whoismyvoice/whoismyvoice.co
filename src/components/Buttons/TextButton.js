import React, { Component, } from 'react';
import PropTypes from 'prop-types';

// Styles
import '../../styles/TextButton.css';

class TextButton extends Component {
  render() {
    const { link, text, } = this.props;
    let buttonClasses = 'text-button';
    if (text === 'Back') {
      buttonClasses = buttonClasses + 'text-button--back';
    }
    return <a href={link} className={buttonClasses} onClick={this.props.onClick}>
      {text}
      <span className="text-button-border">
      </span>
  	</a>;
  }
}

TextButton.propTypes = {
  link: PropTypes.string,
  text: PropTypes.string,
};

export default TextButton;
