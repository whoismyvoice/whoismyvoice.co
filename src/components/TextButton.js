import React from 'react';
import { Link } from 'react-router';
import SenateActions from '../actions/SenateActions';

// Component
import BaseComponent from './BaseComponent';

// Styles
import style from './../styles/Button.scss';

class TextButton extends BaseComponent {
  constructor() {
    super();
    this._bind('_destroyFullpage', '_handleRestart');
  }

  _destroyFullpage() {
    if ($.fn.fullpage.destroy !== undefined) {
      $.fn.fullpage.destroy();
    }
  }

  _handleRestart() {
    SenateActions.flush();
    this._destroyFullpage();
  }

  render() {
    const {link, flush, text} = this.props;

    return <a href={link} className="text-button" onClick={this.props.onClick}>
      <div className="text">
        {text}
      </div>
      <div className="border">
      </div>
  	</a>;
  }
};

TextButton.propTypes = {
  flush: React.PropTypes.bool,
  link: React.PropTypes.string,
  text: React.PropTypes.string,
};

export default TextButton;
