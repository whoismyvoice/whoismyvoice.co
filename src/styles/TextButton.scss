@import 'variables';

.text-button {
  position: relative;
  display: block;
  margin: 0 auto;
  margin-top: 30px;
  letter-spacing: 1px;
  padding-bottom: 5px;
  cursor: pointer;
  padding: 5px;
  width: 140px;
  color: $black;
  text-transform: uppercase;
  font-weight: 600;
}

.text-button-border {
  display: block;
  margin: 0 auto;
  margin-top: 5px;
  width: 50%;
  height: 2px;
  background-color: $black;
}
