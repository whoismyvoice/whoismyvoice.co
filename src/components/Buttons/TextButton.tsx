import React from 'react';
import cx from 'classnames';

// Styles
import '../../styles/TextButton.css';

interface Props {
  link?: string;
  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
  text: string;
}

interface State {
  buttonClasses: string;
}

class TextButton extends React.Component<Props, State> {
  static defaultProps = {
    onClick: () => undefined,
  };

  state = {
    buttonClasses: 'text-button',
  };

  constructor(props: Props) {
    super(props);
    const { text } = props;
    this.state = {
      buttonClasses: this.getButtonClasses(text),
    };
  }

  getButtonClasses(text: string) {
    return cx('text-button', {
      'text-button--back': text === 'Back',
    });
  }

  UNSAFE_componentWillReceiveProps(nextProps: Readonly<Props>) {
    const { text } = this.props;
    const nextText = nextProps.text;
    if (text !== nextText) {
      this.setState({
        buttonClasses: this.getButtonClasses(nextText),
      });
    }
  }

  render() {
    const { link, onClick, text } = this.props;
    const { buttonClasses } = this.state;
    return (
      <a href={link} className={buttonClasses} onClick={onClick}>
        {text}
        <span className="text-button-border" />
      </a>
    );
  }
}

export default TextButton;
