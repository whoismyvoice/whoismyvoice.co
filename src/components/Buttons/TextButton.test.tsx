import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';
import TextButton from './TextButton';

const props = {
  link: 'link',
  text: 'text',
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TextButton {...props} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer.create(<TextButton {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
