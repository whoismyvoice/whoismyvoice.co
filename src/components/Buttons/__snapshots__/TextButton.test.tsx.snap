// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="text-button"
  href="link"
  onClick={[Function]}
>
  text
  <span
    className="text-button-border"
  />
</a>
`;
